src/FlaUILibrary/flaui/module/mouse.py

```diff
@@ -147,3 +178,51 @@ def _drag_and_drop(element_from: Any, element_to: Any):
             FlaUI.Core.Input.Mouse.Drag(element_from.GetClickablePoint(), element_to.GetClickablePoint())
         except NoClickablePointException:
             raise FlaUiError(FlaUiError.ElementNotClickable) from None
+
+    def _click_open(self, click_type, click_element_xpath: str, open_element_xpath: str, 
+                    focus_element_xpath_before_click: str = None, focus_element_xpath_after_open: str = None , 
+                    max_repeat: int = 5, timeout_between_repeates: int = 1000, ignore_if_already_open: bool = True):
+        try:
+            if ignore_if_already_open:
+                alreadyOpen = self._automation.GetDesktop().FindFirstByXPath(open_element_xpath)
+                if alreadyOpen:
+                    return True
+            if focus_element_xpath_before_click:
+                self._automation.GetDesktop().FindFirstByXPath(focus_element_xpath_before_click).focus()
+            for i in range(max_repeat):
+                click_element = self._automation.GetDesktop().FindFirstByXPath(click_element_xpath)
+                if click_element:
+                    click_type(click_element)
+                    openElement = self._automation.GetDesktop().FindFirstByXPath(open_element_xpath)
+                    if openElement:
+                        if focus_element_xpath_after_open:
+                            self._automation.GetDesktop().FindFirstByXPath(focus_element_xpath_after_open).focus()
+                        return True
+                time.sleep(float(timeout_between_repeates)/1000)
+            raise FlaUiError(FlaUiError.ElementNotOpened.format(open_element_xpath, click_element_xpath))
+        except NoClickablePointException:
+            raise FlaUiError(FlaUiError.ElementNotClickable) from None
+
+    def _click_close(self, click_type, click_element_xpath: str, close_element_xpath: str, 
+                    focus_element_xpath_before_click: str = None, focus_element_xpath_after_close: str = None , 
+                    max_repeat: int = 5, timeout_between_repeates: int = 1000, ignore_if_already_close: bool = True):
+        try:
+            if ignore_if_already_close:
+                alreadyOpen = self._automation.GetDesktop().FindFirstByXPath(close_element_xpath)

```

Comment:
You should not handle here any direct find operation. All elements should be send as parameter.

Get Element Keyword should be used in Keyword side to receive all keywords. Because this usage does not handle any error handling if element does not exists.